### PR TITLE
Fetch seen labels from the database

### PR DESCRIPTION
**Pull request by badboy**
This goes through the database and looks for all stored data for the
named labeled metric and extracts the labels.

I thought about extracting the extraction part into a function but couldn't come up with a proper name for it.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/src/metrics/labeled.rs [hunk 1]
```diff
@@ -67,8 +69,30 @@ where
...
+
+            let lifetime = self.meta.lifetime;
+            for store in &self.meta.send_in_pings {
+                glean
+                    .storage()
+                    .iter_store_from(lifetime, store, &mut snapshotter);
+            }
+
+            // Swap the newly seen labels with the old (empty) storage.
+            mem::swap(&mut self.seen_labels, &mut seen_labels);
```
Comment (mdboom) [inline glean-core/src/metrics/labeled.rs]: Learning question: Is this just so this is transactional, or is there a ownership reason to do this?

Comment (badboy) [reply glean-core/src/metrics/labeled.rs]: Actually ... it was for ownership reasons, but I now see how to refactor it.

**Comment by badboy** [comment]
Failed because:

```
            > Could not GET 'https://jcenter.bintray.com/com/squareup/javawriter/2.5.0/javawriter-2.5.0.pom'. Received status code 504 from server: Gateway Time-out
```

I rekicked the test.

File: glean-core/src/metrics/labeled.rs [hunk 2]
```diff
@@ -67,8 +68,27 @@ where
         }
     }
 
-    fn dynamic_label<'a>(&mut self, label: &'a str) -> &'a str {
-        // TODO(bug 1554970): Fetch seen_labels from the database if empty
+    fn dynamic_label<'a>(&mut self, glean: &Glean, label: &'a str) -> &'a str {
```
Comment (Dexterp37) [inline glean-core/src/metrics/labeled.rs]: I know this isn't a public function, but can you please add documentation for what this is supposed to do?

File: glean-core/tests/labeled.rs [hunk 3]
```diff
@@ -251,3 +252,70 @@ fn dynamic_labels_regex_mimsatch() {
         snapshot
     );
 }
+
+#[test]
+fn seen_labels_get_reloaded_from_disk() {
```
Comment (Dexterp37) [inline glean-core/tests/labeled.rs]: Will this allow us to enable [this test](https://github.com/mozilla/glean/blob/93736f8598b607f77c96bf9cda266c7716dde987/glean-core/android/src/test/java/mozilla/telemetry/glean/private/LabeledMetricTypeTest.kt#L436) again?

Comment (badboy) [reply glean-core/tests/labeled.rs]: once I understand the mocking part

Comment (Dexterp37) [reply glean-core/tests/labeled.rs]: I can help you with that, no worries :) ping me about i!

Comment (badboy) [reply glean-core/tests/labeled.rs]: I took a look and can't really see how to re-enable that tests (or even that we _should_ re-enable it):

* It's mocking sharedpreferences, which we are not using anymore
* We can't easily inject data into the store, without going through the rudst part anyway.
* This case is covered by the Rust test

How about removing that kotlin test case?
wdyt, @Dexterp37?

Comment (Dexterp37) [reply glean-core/tests/labeled.rs]: I think you're right. Let's remove the test and add something like this in its place:

```
// SKIPPED `test seen labels get reloaded from disk`
// REASON XYZ HERE
```

(here or in a follow-up)

**Pull request merged**